sky/utils/config_utils.py

```diff
@@ -199,6 +199,15 @@ def merge_k8s_configs(
                         else:
                             base_config[key].append(new_volume)
             else:
-                base_config[key].extend(value)
+                # For other list values, merge lists and maintain uniqueness.
+                # Preserve the order - first list is the base_config list, and
+                # the second list is the override list. This is required for
+                # order sensitive lists like allowed_contexts.
+                seen = set(base_config[key])
```

Comment:
Is it possible that the list items are dicts (unhashable)?

Reply:
oh yeah good point - I was thinking they'd be covered in the special casing above (e.g., volumes), but realized there may be other keys and it's good to be general. Simplified the logic here to also work with dicts.

Reply:
Also added relevant test cases.